content/documentation/integration_maximizing_transaction_size.mdx

```diff
@@ -0,0 +1,82 @@
+---
+title: Maximizing Transaction Size
+description: Integration Guide | Maximizing Transaction Size | Iron Fish Documentation
+---
+
+Iron Fish transactions, in their simplest form, spend notes to fund payments (plus the miners fee).
+The Iron Fish [block](/use/get-started/glossary#Block) has a predefined size limit, so there is a limit on the size and number of transactions in a block.
+In effect, if you are maximizing transaction size, you are trying to make the transaction size equal to the max
+block size.
+
+Here's an example that creates a transaction with a rough estimate of the number of notes that can fit in a block, then incrementally removes notes until the transaction will fit in a block.
+
+```js
+import { Asset } from '@ironfish/rust-nodejs'
+import {
+  CreateTransactionRequest,
+  CurrencyUtils,
+  IronfishSdk,
+  RawTransactionSerde,
+} from '@ironfish/sdk'
+import { NOTE_ENCRYPTED_SERIALIZED_SIZE_IN_BYTE } from '@ironfish/sdk/src/primitives/noteEncrypted'
+async function main(): Promise<void> {
+  // creating fake outputs for example
+  const outputs: CreateTransactionRequest['outputs'] = []
+  for (let i = 0; i < 100000; i++) {
+    outputs.push({
+      publicAddress: 'd841fcda692fdc030e772037c27c5c8d57b4d55355f8636126f338b74d59de1c',
+      amount: CurrencyUtils.encode(1n),
+      memo: 'test',
+      assetId: Asset.nativeId().toString('hex'),
+    })
+  }
+
+  const sdk = await IronfishSdk.init({ dataDir: '~/.dev0' })
+  const client = await sdk.connectRpc()
+  // Get the max block size from the node
+  const consensusParams = await client.chain.getConsensusParameters()
+  const maxBlockSizeBytes = consensusParams.content.maxBlockSizeBytes
```

Comment:
Transactions are rejected if greater than `this.chain.consensus.parameters.maxBlockSizeBytes - getBlockWithMinersFeeSize()` -- I don't think that's exported yet from the SDK

https://github.com/iron-fish/ironfish/blob/6dc93a47c2bb67e2fb8634016e9b2998241d1ca1/ironfish/src/consensus/verifier.ts#L267

Reply:
Since it's not exported in the SDK, we can merge this PR without the change and update the doc when this is exported from the SDK